pkg/rke2/machine_filters.go

```diff
@@ -55,3 +113,21 @@ func matchesTemplateClonedFrom(infraConfigs map[string]*unstructured.Unstructure
 		return true
 	}
 }
+
+// matchesKubernetesVersion returns a filter to find all machines that match a given Kubernetes version.
+func matchesKubernetesVersion(kubernetesVersion string) func(*clusterv1.Machine) bool {
+	return func(machine *clusterv1.Machine) bool {
+		if machine == nil {
+			return false
+		}
+		if machine.Spec.Version == nil {
+			return false
+		}
+
+		rcpKubeVersion, err := bsutil.Rke2ToKubeVersion(kubernetesVersion)
+		if err != nil {
+			return true
```

Comment:
We'll loose the error here. Not sure what can be done about that.

Reply:
This is inside a matching func, for which the error should be ignored, or considered as an unimportant criteria to match against. However, there are other places where using `bsutil.Rke2ToKubeVersion(rke2Version)` will cause an error!